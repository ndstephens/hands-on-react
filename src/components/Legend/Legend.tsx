import styled from '@emotion/styled';

export const Legend: React.FC = () => (
  <Parent>
    <strong>flag: </strong>
    <Action>
      <Key>ctrl</Key> + <Click>click</Click>
    </Action>
  </Parent>
);

export default Legend;

const Parent = styled.legend`
  font-size: 1em;
  margin: 0 auto 2vw;
  line-height: 1.25em;
`;

const Action = styled.code`
  background-color: #e3e3e3;
`;

const Key = styled.span`
  color: #ec433c;
`;

const Click = styled.span`
  color: #2a48ec;
`;
