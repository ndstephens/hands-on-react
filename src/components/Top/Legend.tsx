import styled from '@emotion/styled';

export interface LegendProps {
  /**
   * Feature that should be activated after first & second actions
   */
  feature: string;
  /**
   * First action
   */
  firstAction: string;
  /**
   * Second Action
   */
  secondAction: string;
}

export const Legend: React.FC<LegendProps> = ({
  feature,
  firstAction,
  secondAction,
}) => (
  <Parent>
    <strong>{feature}: </strong>
    <Actions>
      <FirstAction>{firstAction}</FirstAction> +{' '}
      <SecondAction>{secondAction}</SecondAction>
    </Actions>
  </Parent>
);

const Parent = styled.legend`
  font-size: 1em;
  margin: 0 auto 2vw;
  line-height: 1.25em;
`;

const Actions = styled.code`
  background-color: #e3e3e3;
`;

const FirstAction = styled.span`
  color: #ec433c;
`;

const SecondAction = styled.span`
  color: #2a48ec;
`;
