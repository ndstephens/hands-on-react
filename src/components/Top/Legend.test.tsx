import { render } from '@testing-library/react';

import { Legend } from './Legend';

it('Legend renders correctly', () => {
  const { asFragment } = render(
    <Legend feature="Flag" firstAction="ctrl" secondAction="click" />
  );

  expect(asFragment()).toMatchSnapshot();
});
