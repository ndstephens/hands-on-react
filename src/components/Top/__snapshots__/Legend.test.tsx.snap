// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend renders correctly 1`] = `
<DocumentFragment>
  <legend
    class="css-17qz6a2"
  >
    <strong>
      Flag: 
    </strong>
    <code
      class="css-h9xasx"
    >
      <span
        class="css-udfepp"
      >
        ctrl
      </span>
       + 
      <span
        class="css-9c23k1"
      >
        click
      </span>
    </code>
  </legend>
</DocumentFragment>
`;
